src/Sarif.FunctionalTests/SpecExamples/Comprehensive.sarif

```diff
@@ -74,42 +82,153 @@
           "formattedRuleMessage": {
             "formatId": "default",
             "arguments": [
-              "n"
+              "p"
             ]
           },
+          "suppressionStates": [ "suppressedExternally" ],
+          "baselineState": "existing",
+          "level": "error",
+          "snippet": "add_core(p, offset, val);\n    return;",
           "locations": [
             {
               "analysisTarget": {
-                "uri": "collections/list.cpp",
+                "uri": "file:///home/build/src/collections/list.cpp"
+              },
+              "resultFile": {
+                "uri": "file:///home/build/src/collections/list.h",
                 "region": {
-                  "startLine": 15
+                  "startLine": 15,
+                  "startColumn": 9,
+                  "endLine": 15,
+                  "endColumn": 10,
+                  "length": 1,
+                  "offset": 254
+                }
+              },
+              "fullyQualifiedLogicalName": "collections::list:add",
+              "decoratedName": "?add@list@collections@@QAEXH@Z"
+            }
+          ],
+          "relatedLocations": [
+            {
+              "message": "\"count\" was declared here.",
+              "physicalLocation": {
+                "uri": "file:///home/build/src/collections/list.h",
+                "region": {
+                  "startLine": 8,
+                  "startColumn": 5
                 }
               },
               "fullyQualifiedLogicalName": "collections::list:add"
             }
           ],
           "codeFlows": [
             {
-              "message": "Code flow #1",
+              "message": "Path from declaration to usage",
               "locations": [
                 {
                   "step": 0,
                   "kind": "declaration",
                   "importance": "essential",
-                  "fullyQualifiedLogicalName": "collections::list::add",
+                  "message": "Variable \"p\" declared.",
```

Comment:
We shouldn't ignore the weight of MS usage. One strong reason in our context (reporting messages) for favoring single quotes is that static analysis messages themselves are extracted and quoted in other contexts.